### PR TITLE
Refactored to support device info function

### PR DESCRIPTION
**Pull request by milanperera**
Did changes to both android client and jaxrs

### CODE REVIEW

File: modules/mobile-agents/android/client/src/org/wso2/mdm/agent/services/BuildDeviceInfoPayload.java [hunk 1]
```diff
@@ -20,90 +20,212 @@
...
+			Log.e(TAG, errorMsg);
+			throw new AndroidAgentException(errorMsg, e);
+		}
+		property = new Device.Property();
+		property.setName(Constants.Device.INFO);
+		property.setValue(deviceInfoPayload);
+		properties.add(property);
+
+		device.setProperties(properties);
+
```
Comment (DilanUA) [inline modules/mobile-agents/android/client/src/org/wso2/mdm/agent/services/BuildDeviceInfoPayload.java]: unnecessary empty line

File: modules/mobile-agents/android/client/src/org/wso2/mdm/agent/services/Operation.java [hunk 2]
```diff
@@ -186,54 +178,33 @@ public void doTask(org.wso2.mdm.agent.beans.Operation operation) throws AndroidA
...
-			}
+		String replyPayload = deviceInfoPayload.getDeviceInfoPayload();
 
-		} catch (JSONException e) {
-			operation.setStatus(resources.getString(R.string.operation_value_error));
-			resultBuilder.build(operation);
-			throw new AndroidAgentException("Invalid JSON format.", e);
-		}
+		String ipSaved =
+				Preference.getString(context.getApplicationContext(),
```
Comment (DilanUA) [inline modules/mobile-agents/android/client/src/org/wso2/mdm/agent/services/Operation.java]: improper line break

File: modules/mobile-agents/android/client/src/org/wso2/mdm/agent/services/Operation.java [hunk 3]
```diff
@@ -186,54 +178,33 @@ public void doTask(org.wso2.mdm.agent.beans.Operation operation) throws AndroidA
...
+		ServerConfig utils = new ServerConfig();
+		utils.setServerIP(ipSaved);
+
+		String url = utils.getAPIServerURL() + Constants.DEVICE_ENDPOINT + deviceInfo.getMACAddress();
+
+		CommonUtils.callSecuredAPI(context, url,
+				org.wso2.mdm.agent.proxy.utils.Constants.HTTP_METHODS.PUT, replyPayload, 
+				Operation.this,
+				Constants.DEVICE_INFO_REQUEST_CODE
+		);
```
Comment (DilanUA) [inline modules/mobile-agents/android/client/src/org/wso2/mdm/agent/services/Operation.java]: improper indentation

File: modules/mobile-agents/android/client/src/org/wso2/mdm/agent/services/Operation.java [hunk 4]
```diff
@@ -1022,7 +995,39 @@ private void installApplication(JSONObject data, org.wso2.mdm.agent.beans.Operat
 		}
 	}
 
+	/**
+	 * This method returns the completed operations list
```
Comment (DilanUA) [inline modules/mobile-agents/android/client/src/org/wso2/mdm/agent/services/Operation.java]: no . sign to mark the end of method description

File: modules/mobile-agents/android/client/src/org/wso2/mdm/agent/services/Operation.java [hunk 5]
```diff
@@ -1022,7 +995,39 @@ private void installApplication(JSONObject data, org.wso2.mdm.agent.beans.Operat
...
+		if (requestCode == Constants.DEVICE_INFO_REQUEST_CODE) {
+			if (result != null) {
+				responseStatus = result.get(Constants.STATUS_KEY);
+				if (Constants.REQUEST_SUCCESSFUL.equals(responseStatus)) {
+					response = result.get(Constants.RESPONSE);
+					if (response != null && !response.isEmpty()) {
+						if (Constants.DEBUG_MODE_ENABLED) {
+							Log.d(TAG, "onReceiveAPIResult." + response);
+							Log.d(TAG, "Device information sent");
+
```
Comment (DilanUA) [inline modules/mobile-agents/android/client/src/org/wso2/mdm/agent/services/Operation.java]: unnecessary empty line

File: modules/mobile-agents/android/client/src/org/wso2/mdm/agent/services/BuildDeviceInfoPayload.java (hunk 1 above)
Comment (milanperera) [inline modules/mobile-agents/android/client/src/org/wso2/mdm/agent/services/BuildDeviceInfoPayload.java]: Fixed in PR [#70].

File: modules/mobile-agents/android/client/src/org/wso2/mdm/agent/services/Operation.java (hunk 2 above)
Comment (milanperera) [inline modules/mobile-agents/android/client/src/org/wso2/mdm/agent/services/Operation.java]: Fixed in PR [#70].

File: modules/mobile-agents/android/client/src/org/wso2/mdm/agent/services/Operation.java (hunk 3 above)
Comment (milanperera) [inline modules/mobile-agents/android/client/src/org/wso2/mdm/agent/services/Operation.java]: Fixed in PR [#70].

File: modules/mobile-agents/android/client/src/org/wso2/mdm/agent/services/Operation.java (hunk 4 above)
Comment (milanperera) [inline modules/mobile-agents/android/client/src/org/wso2/mdm/agent/services/Operation.java]: Fixed in PR [#70].

File: modules/mobile-agents/android/client/src/org/wso2/mdm/agent/services/Operation.java (hunk 5 above)
Comment (milanperera) [inline modules/mobile-agents/android/client/src/org/wso2/mdm/agent/services/Operation.java]: Fixed in PR [#70].

**Pull request merged**